File: app/src/main/java/com/widiarifki/screentest/adapter/EventAdapter.java
package com.widiarifki.screentest.adapter;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.widiarifki.screentest.MainActivity;
import com.widiarifki.screentest.R;
import com.widiarifki.screentest.model.Event;
import com.widiarifki.screentest.presentation.menu.MenuFragment;

import java.util.List;

/**
 * Created by widiarifki on 24/02/2018.
 */

public class EventAdapter extends RecyclerView.Adapter {

    List<Event> mObjects;
    Context mContext;

    public EventAdapter(Context context, List<Event> objects){
        mContext = context;
        mObjects = objects;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(final ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.item_list_event, null);
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int position = ((RecyclerView)parent).getChildLayoutPosition(v);
                Event eventData = mObjects.get(position);

                Bundle args = new Bundle();
                args.putString("EVENT", eventData.getNama());
                Fragment secondFrag = new MenuFragment();
                secondFrag.setArguments(args);
                ((MainActivity)mContext).setFragment(secondFrag, MenuFragment.TITLE);
            }
        });
        return new EventViewHolder(view);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        Event dataItem = mObjects.get(position);
        EventViewHolder viewHolder = (EventViewHolder) holder;
        viewHolder.txtEvent.setText(dataItem.getNama());
        viewHolder.txtEventDate.setText(dataItem.getTanggal());
        viewHolder.txtDesc.setText(dataItem.getDesc());
        viewHolder.imgEvent.setImageResource(dataItem.getImgResId());
    }

    @Override
    public int getItemCount() {
        return mObjects.size();
    }

    private class EventViewHolder extends RecyclerView.ViewHolder {
        ImageView imgEvent;
        TextView txtEvent;
        TextView txtEventDate;
        TextView txtDesc;

        public EventViewHolder(View view) {
            super(view);
            imgEvent = (ImageView) view.findViewById(R.id.imgEvent);
            txtEvent = (TextView) view.findViewById(R.id.txtEvent);
            txtEventDate = (TextView) view.findViewById(R.id.txtEventDate);
            txtDesc = (TextView) view.findViewById(R.id.txtDesc);
        }
    }
}

File: app/src/main/java/com/widiarifki/screentest/adapter/EventAdapter.kt
package com.widiarifki.screentest.adapter

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.widiarifki.screentest.Constant
import com.widiarifki.screentest.R
import com.widiarifki.screentest.model.Event
import kotlinx.android.synthetic.main.item_list_event.view.*

/**
 * Created by widiarifki on 24/02/2018.
 */

class EventAdapter(internal var mContext: Context?, internal var mObjects: List<Event>) : RecyclerView.Adapter<EventAdapter.EventViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EventViewHolder? {
        val view = LayoutInflater.from(mContext).inflate(R.layout.item_list_event, null)
        return EventViewHolder(view)
    }

    override fun onBindViewHolder(holder: EventViewHolder?, position: Int) {
        val dataItem = mObjects[position]
        holder?.bind(dataItem, position)
    }

    override fun getItemCount(): Int {
        return mObjects.size
    }

    inner class EventViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        public fun bind(event: Event, position: Int){
            itemView.txtEvent.text = event.nama
            itemView.txtEventDate.text = event.tanggal
            itemView.txtDesc.text = event.desc
            itemView.imgEvent.setImageResource(event.imgResId)
            itemView.setOnClickListener { v ->
                val intent = Intent()
                intent.putExtra(Constant.extraTagEventName, event.nama)
                (mContext as Activity).setResult(Activity.RESULT_OK, intent)
                (mContext as Activity).finish()
            }
        }
    }
}
